test/rekt/resources/pod/pod.yaml

```diff
@@ -16,7 +16,11 @@ apiVersion: v1
 kind: Pod
 metadata:
   name: {{ .name }}
+  {{ if .overriddenNamespace }}
+  namespace: {{ .overriddenNamespace }}
+  {{ else }}
   namespace: {{ .namespace }}
+  {{ end }}
```

Comment:
why the template should care about `overriddenNamespace` vs `namespace`?
Can we keep using namespace _only_?

Reply:
Namespace is the current test's namespace. For this test, we've had to create other temporary namespaces and create resources in them. The overridden namespace is there in order to create resources in a namespace other than the one the test is running in.

Reply:
But this doesn't seem the correct layer to fix that limitation of the rekt framework since injecting a `namespace` variable would anyway set the correct namespace we expect without having another variable `overriddenNamespace`. 

I get that `namespace` is forced to be the test namespace but that seem wrong to me if the user intention is to inject a specific namespace, so I'd fix it in rekt with something like this in reconciler-test/pkg/environment/magic.go:

```
	if _, ok := cfg["namespace"]; !ok {
       	    cfg["namespace"] = mr.namespace
        }
```

otherwise if any other resource needs something like this, we end up with templates that very complicated for no reason

Reply:
Sounds good. I wanted to avoid `reconciler-test` changes since this use case seemed like a one-off. I do prefer this solution